tests/plotting/test_widgets.py

```diff
@@ -2,136 +2,170 @@
 import pytest
 import vtk
 
-import pyvista
+import pyvista as pv
 from pyvista import examples
+from pyvista.core.errors import VTKVersionError
+from pyvista.plotting.affine_widget import DARK_YELLOW, get_angle, ray_plane_intersection
 
 # skip all tests if unable to render
 pytestmark = pytest.mark.skip_plotting
 
 
+def r_mat_to_euler_angles(R):
+    """
+    Extract Euler angles from a 3x3 rotation matrix using the ZYX sequence.
+    Returns the angles in radians.
+    """
+
+    # Check for gimbal lock: singular cases
+    if abs(R[2, 0]) == 1:
+        # Gimbal lock exists
+        yaw = 0  # Set yaw to 0 and compute the others
+        if R[2, 0] == -1:  # Directly looking up
+            pitch = np.pi / 2
+            roll = yaw + np.arctan2(R[0, 1], R[0, 2])
+        else:  # Directly looking down
+            pitch = -np.pi / 2
+            roll = -yaw + np.arctan2(-R[0, 1], -R[0, 2])
+    else:
+        # Not at the singularities
+        pitch = -np.arcsin(R[2, 0])
+        roll = np.arctan2(R[2, 1] / np.cos(pitch), R[2, 2] / np.cos(pitch))
+        yaw = np.arctan2(R[1, 0] / np.cos(pitch), R[0, 0] / np.cos(pitch))
+
+    if np.isclose(roll, np.pi):
+        roll = 0
+    if np.isclose(pitch, np.pi):
+        pitch = 0
+    if np.isclose(yaw, np.pi):
+        yaw = 0
+
+    return roll, pitch, yaw
+
+
 def test_widget_box(uniform):
-    p = pyvista.Plotter()
+    p = pv.Plotter()
```

Comment:
@akaszynski Could you please merge #4805 first to eliminate these irrelevant differences? Since #4805 is only a change in the test code, you can merge it immediately (I mean we don't need the 24-hour rule).